web/src/scheduled_messages.js

```diff
@@ -273,9 +274,12 @@ export function get_filtered_send_opts(date) {
 
     let possible_send_later_today = {};
     let possible_send_later_monday = {};
-    if (hours <= 8) {
+    // Show Today send options based on time of day
+    if (hhmm <= 859 - MINIMUM_SCHEDULED_MESSAGE_DELAY_SECONDS / 60) {
```

Comment:
Why not just write this `hhmm < 900 - MINIMUM_SCHEDULED_MESSAGE_DELAY_SECONDS / 60`?

Reply:
Oh, because it'd need to be 860 to do that. Yeah I think we probably don't want to do that; it's kinda evil. I made the following changes before merging this:

```
diff --git a/web/src/scheduled_messages.js b/web/src/scheduled_messages.js
index 8c8a3abe20..6ab1a29cae 100644
--- a/web/src/scheduled_messages.js
+++ b/web/src/scheduled_messages.js
@@ -196,8 +196,6 @@ export function get_filtered_send_opts(date) {
     const send_times = compute_send_times(date);
 
     const day = date.getDay(); // Starts with 0 for Sunday.
-    // Construct an HHMM integer of the time for comparison below
-    const hhmm = date.getHours() * 100 + date.getMinutes(); // e.g., 1806 at 6:06pm
 
     const send_later_today = {
         today_nine_am: {
@@ -274,11 +272,13 @@ export function get_filtered_send_opts(date) {
 
     let possible_send_later_today = {};
     let possible_send_later_monday = {};
+
+    const minutes_into_day = date.getHours() * 60 + date.getMinutes();
     // Show Today send options based on time of day
-    if (hhmm <= 859 - MINIMUM_SCHEDULED_MESSAGE_DELAY_SECONDS / 60) {
+    if (minutes_into_day < 9 * 60 - MINIMUM_SCHEDULED_MESSAGE_DELAY_SECONDS / 60) {
         // Allow Today at 9:00am only up to minimum scheduled message delay
         possible_send_later_today = send_later_today;
-    } else if (hhmm <= 1359 - MINIMUM_SCHEDULED_MESSAGE_DELAY_SECONDS / 60) {
+    } else if (minutes_into_day < (12 + 4) * 60 - MINIMUM_SCHEDULED_MESSAGE_DELAY_SECONDS / 60) {
         // Allow Today at 4:00pm only up to minimum scheduled message delay
         possible_send_later_today.today_four_pm = send_later_today.today_four_pm;
     } else {
diff --git a/web/tests/scheduled_messages.test.js b/web/tests/scheduled_messages.test.js
index 8b63d21397..efe26741de 100644
--- a/web/tests/scheduled_messages.test.js
+++ b/web/tests/scheduled_messages.test.js
@@ -121,8 +121,8 @@ run_test("scheduled_modal_opts", () => {
         {hour: "T08:54:00", extras: ["today_nine_am", "today_four_pm"]},
         {hour: "T08:57:00", extras: ["today_four_pm"]},
         {hour: "T11:00:00", extras: ["today_four_pm"]},
-        {hour: "T13:54:00", extras: ["today_four_pm"]},
-        {hour: "T13:57:00", extras: []},
+        {hour: "T15:54:00", extras: ["today_four_pm"]},
+        {hour: "T15:57:00", extras: []},
         {hour: "T17:00:00", extras: []},
     ];
 
```